schema-guide.md

```diff
@@ -533,11 +590,9 @@ TODO explain why we use definitions
 
 ### `definitions.date`
 
-`definitions` and its subkeys should not be used directly in `CITATION.cff` files.
-
-- **type**: `string`
+- **type**: Nonempty `string`
 - **required**: `false`
-- **description**: A date. Format is 4-digit year, 2-digit month, 2-digit day of month, separated by dashes.
+- **description**: A date. Format is 4-digit year, followed by 2-digit month, followed by 2-digit day of month, separated by dashes, and double-quoted or unquoted.
```

Comment:
```suggestion
- **description**: A date. Format is 4-digit year, followed by 2-digit month, followed by 2-digit day of month, separated by dashes, and double-quoted.
```

Reply:
But they can be unquoted! Then would be YAML timestamps in a specific format. This is why we parse to string in schema_poc.py and ruby-cff. I'd rather not mention quotes at all than forcing the user to quote dates?

Reply:
What do you think @jspaaks?

Reply:
```suggestion
- **description**: A date. Format is 4-digit year, followed by 2-digit month, followed by 2-digit day of month, and separated by dashes.
```

As agreed via messenger, we won't mention quotes here.